packages/protocol/contracts/L1/provers/GuardianProver.sol

```diff
@@ -3,23 +3,37 @@ pragma solidity 0.8.24;
 
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
 import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
-
+import "../../common/EssentialContract.sol";
 import "../../common/LibStrings.sol";
 import "../../verifiers/IVerifier.sol";
-import "../tiers/ITierProvider.sol";
 import "../ITaikoL1.sol";
-import "./Guardians.sol";
 
 /// @title GuardianProver
 /// This prover uses itself as the verifier.
 /// @custom:security-contact security@taiko.xyz
-contract GuardianProver is IVerifier, Guardians {
+contract GuardianProver is IVerifier, EssentialContract {
     using SafeERC20 for IERC20;
 
-    error GV_PERMISSION_DENIED();
-    error GV_ZERO_ADDRESS();
+    /// @notice Contains the index of the guardian in `guardians` plus one (zero means not a
+    /// guardian)
+    /// @dev Slot 1
+    mapping(address guardian => uint256 id) public guardianIds;
+
+    /// @notice Mapping to store the approvals for a given hash, for a given version
+    mapping(uint32 version => mapping(bytes32 hash => uint256 approvalBits)) internal _approvals;
+
+    /// @notice The set of guardians
+    /// @dev Slot 3
+    address[] public guardians;
+
+    /// @notice The version of the guardians
+    /// @dev Slot 4
+    uint32 public version;
 
-    uint256[50] private __gap;
+    /// @notice The minimum number of guardians required to approve
+    uint32 public minGuardians;
+
+    uint256[46] private __gap;
```

Comment:
Hmm, will we be fully storage compatible if we leave out `GuardianProver` contract's 50 uint256-long slot ? I guess so since they were unsued, just flagging here.

Reply:
Yes, fully compatiable as long as there is no sub-contract of this contract deployed/used.